packages/crawler/src/crawl-websites.ts

```diff
@@ -0,0 +1,95 @@
+import { crawlerConfig as config } from '../constants';
+import * as puppeteer from 'puppeteer';
+import { evaluatePage } from './evaluate-page';
+import {
+  getAggregatedConsumedTagNames,
+  getAggregatedConsumedTagNamesForVersionsAndPrefixes,
+  getConsumedPrefixesForVersions,
+  getPdsTagNamesWithPropertyNames,
+  getRawDataWithoutVersionsAndPrefixes,
+} from './helpers/convert-data-helper';
+import { writeGeneralReport, writeWebsiteReport } from './helpers/fs-helper';
+import { ConsumedTagNamesForVersionsAndPrefixes, TagNameData, TagNamesWithPropertyNames } from './types';
+import { Page } from 'puppeteer';
+
+export const crawlPage = async (page: Page, websiteUrl: string): Promise<ConsumedTagNamesForVersionsAndPrefixes> => {
+  console.log('Crawling page ' + websiteUrl);
+  // getting raw data
+  return await evaluatePage(page, getPdsTagNamesWithPropertyNames());
+};
+export const crawlWebsite = async (browser: puppeteer.Browser, websiteUrl: string): Promise<TagNameData[]> => {
+  const page = await browser.newPage();
+  // at least porsche finder seems to check the headers to block scrapers, setting the UA solves this
+  await page.setUserAgent(
+    'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36'
+  );
+  await page.goto(websiteUrl, {
+    waitUntil: 'networkidle0',
+  });
+
+  const pdsCrawlerRawData = await crawlPage(page, websiteUrl);
+
+  // raw data in another format - without versions and prefixes
+  const pdsCrawlerRawDataWithoutVersionsAndPrefixes = getRawDataWithoutVersionsAndPrefixes(pdsCrawlerRawData);
+  console.log('Aggregating data for ' + page.url());
+  // info about used versions and prefixes
+  const consumedPdsVersionsWithPrefixes = getConsumedPrefixesForVersions(pdsCrawlerRawData);
+
+  // aggregated data
+  const aggregatedConsumedTagNamesForVersionsAndPrefixes =
+    getAggregatedConsumedTagNamesForVersionsAndPrefixes(pdsCrawlerRawData);
+  // aggregated data without versions and prefixes
+  const aggregatedConsumedTagNames = getAggregatedConsumedTagNames(pdsCrawlerRawDataWithoutVersionsAndPrefixes);
+
+  writeWebsiteReport(
+    websiteUrl,
+    JSON.stringify(
```

Comment:
Perhaps a util for this would make sense since the config is duplicated multiple times.

Reply:
Added util for this